### PR TITLE
SH-390 Group Search Api

### PR DESCRIPTION
**Pull request by sknirmalkar89**

### CODE REVIEW

File: group-actors/src/test/java/org/sunbird/actors/SearchGroupActorTest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+    PowerMockito.mockStatic(Localizer.class);
+    when(Localizer.getInstance()).thenReturn(null);
+
+    PowerMockito.mockStatic(ServiceFactory.class);
+    cassandraOperation = mock(CassandraOperationImpl.class);
+    when(ServiceFactory.getInstance()).thenReturn(cassandraOperation);
+  }
+
+  @Test
+  public void testSearchGroup() {
```
Comment (indrajra) [inline group-actors/src/test/java/org/sunbird/actors/SearchGroupActorTest.java:52]: Am proposing to change these test names to make it meaningful. It is difficult to review.
SearchGroupActorTest
   - searchByIdReturnsXYZ
   - searchByInvalidIdReturnsClientError
   - searchWoUserTokenFails
   - searchWMuaTokenReturnsXYZ

File: sb-utils/src/main/java/org/sunbird/util/JsonKey.java [hunk 2]
```diff
@@ -44,4 +41,8 @@
   String TELEMETRY_EVENT_TYPE = "telemetryEventType";
   String PARAMS = "params";
   String API_ACCESS = "api_access";
+  String USER_ID = "userId";
+  String FILTERS = "filters";
+  String COUNT = "count";
```
[the remark this answers is not in the document]
Comment (sknirmalkar89) [reply sb-utils/src/main/java/org/sunbird/util/JsonKey.java:46]: this is count of groups as per the response. let me know if it is not required

File: group-actors/src/main/java/org/sunbird/service/impl/GroupServiceImpl.java [hunk 3]
```diff
@@ -28,9 +32,34 @@ public String createGroup(Group groupObj) throws BaseException {
   }
 
   @Override
-  public Response readGroup(String groupId) throws BaseException{
+  public Response readGroup(String groupId) throws BaseException {
     Response responseObj = groupDao.readGroup(groupId);
     return responseObj;
   }
 
+  @Override
```
[the remark this answers is not in the document]
Comment (sknirmalkar89) [reply group-actors/src/main/java/org/sunbird/service/impl/GroupServiceImpl.java:47]: Added comment

File: sb-utils/src/main/java/org/sunbird/util/JsonKey.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (sknirmalkar89) [reply sb-utils/src/main/java/org/sunbird/util/JsonKey.java]: removed count

**Pull request merged**